test/test_iterative_methods.jl

```diff
@@ -36,3 +36,18 @@ end
 
 # TODO: can I create an input such that newton doesn't converge, but iterative_refinement does?
 # need jacobian to be ill-conditioned...
+
+@testset "large residual warning" begin
+    # a system where bus numbers aren't 1, 2,...is there a smaller one with this property?
+    sys = PSB.build_system(PSB.PSISystems, "RTS_GMLC_DA_sys")
+    data = PowerFlowData(ACPowerFlow(), sys)
+    # First, write solution to data. Then set magnitude of a random bus to a huge number
+    # and try to solve again: the "large residual warning" should be about that bus.
+    solve_powerflow!(data)
+    bus = rand(collect(PSY.get_components(PSY.ACBus, sys)))
```

Comment:
I like my tests to be deterministic, maybe hardcode an index to perturb? Or if you want more than that, loop over the test a few times with random bus based on a hardcoded random seed